scripts/wsrep_sst_xtrabackup-v2.sh

```diff
@@ -1349,11 +1394,28 @@ then
     # This file has the same format as a cnf file.
     #
     sst_info_file_path="${donor_tmpdir}/${SST_INFO_FILE}"
+    transition_key=$(cat /dev/urandom | tr -dc 'a-zA-Z0-9' | fold -w 32 | head -n 1)
     echo "[sst]" > "$sst_info_file_path"
     echo "galera-gtid=$WSREP_SST_OPT_GTID" >> "$sst_info_file_path"
     echo "binlog-name=$(basename "$WSREP_SST_OPT_BINLOG")" >> "$sst_info_file_path"
     echo "mysql-version=$MYSQL_VERSION" >> "$sst_info_file_path"
 
+
+    # append transition_key only if keyring is being used.
+    if [[ $keyring_plugin -eq 1 && $encrypt -ge 0 ]]; then
```

Comment:
Not sure why the "$encrypt -ge 0" test is needed here.  The code below tests for the encryption status.  Also the check here should be "$encrypt -gt 0"

Reply:
Fixed. You are right. -ge 0 is not needed. Check for no presence of encrypted channel is now moved above this block.